### DISCUSSION TITLE
Keine Verbindung OpenDTU zum Wechselrichter HM-800, dieser ist aber an

### DISCUSSION THREAD

**Discussion by OttoKarlHeinz** [tbnobody/OpenDTU#2269] [Q&A] [answered]
Hallo in die Runde,

seit einiger Zeit verwende ich die openDTU (ESP32-D0WD-V3, Ver. 24.8.5 ) zur Nulleinspeisung mit dem HA.
Die openDTU steuert einen HM-400  (bkw_1), an dem eine Solarmodul angeschlossen ist und einen HM-800 (bkw_2), der von einer Batterie Growatt NOAH 2000 gespeist wird.
Die Nulleinspeisung funktioniert bis zur Abschaltung (binary_sensor.bkw_1_producing) des HM-400 für beide Wechselrichter normal.
Nach Abschaltung (ca. 2-5 Minuten ) des HM-400 verliert die openDTU die Verbindung zum HM-400 und zum HM-800 (Letzte Aktualisierung läuft hoch, Soll ist alle 20 Sekunden). 
Nach ca. 1 Stunde aktiviert sich die Verbindung zum HM-800 wieder selber. 
In der Offline-Zeit regelt der HM-800 langsam hoch und langsam herunter (unabhängig von den Daten der Nulleinspeisung, über shelly ablesbar)
Mehrmaliges Neustarten der openDTU führt früher oder später auch zum Ziel.(Online)

Das ganze Verhalten tritt jeden Tag mit dem Abschalten des HM-400 beim Sonnenuntergang auf.

Die OpenDTU und Wechselrichter sind ca. 4m voneinander entfernt. 
Änderungen der NRF24 Sendeleistung beenden nicht die Offline-Zeit (-18 dBm eingestellt).

Im Prinzip ist es das ähnlich wie bei z-Master42
https://github.com/tbnobody/OpenDTU/discussions/1946

nur eben abends.
Woran könnte dieses Verhalten liegen.

VG OttoKarlHeinz

**Comment by stefan123t** [comment, marked as answer]
@OttoKarlHeinz der WR kann der OpenDTU nicht mehr antworten wenn er keinen Strom aus dem NOAH Akku mehr bekommt.
Je nachdem wie und wie oft Deine HA das ActivePowerLimit Command sendet kann man damit die DTU in einen unansprechbaren Zustand bringen, da sie versucht das APL an den WR zu senden und mit dem SystemConfigPara den neuen Limit Wert abfragen muss bevor das entsprechende Command als erfolgreich aus der Priority Queue verschwindet. Wenn dann bereits ein neues APL Command ansteht kommt die DTU da nicht mehr raus.

Bitte ein Logfile anhängen das den Übergang von funktioniert noch alles bis jetzt geht nichts mehr abbildet.

Follow the link to the documentation to setup for USB / serial logging:
https://www.opendtu.solar/firmware/howto/serial_console/

**Reply by OttoKarlHeinz** [reply]
@stefan123t Der WR 2 erhält auch nach dem Sonnenuntergang jeden Tag Strom. Der WR 2 arbeitet nach Abschaltung des WR 1 bei Sonnenuntergang einige Zeit weiter mit dem zuletzt übermittelten ActivePowerLimit weiter, reagiert nicht mehr auf andere Werte, regelt selbständig von Minimum bis Maximum  (30W bis 800W) langsam hin und her. Der Sendezyklus neuer Limits liegt momentan bei 6 Sekunden.
Das funktioniert fast 24h bis auf den Zeitraum der Abschaltung des WR 1 bei Sonnenuntergang. Theoretisch müsste ein größerer Sendezyklus das Problem in der Zeit des Sonnenunterganges lösen?
Den Logfile kann ich erst morgen senden, WR 2 läuft gerade wieder :-)

**Reply by OttoKarlHeinz** [reply]
Habe heute mit Putty das serial logging durchgeführt, leider hat das Abspeichern nicht geklappt. (also morgen nächster Anlauf)
Interessant ist aber, dass der Fehler nicht aufgetreten ist. Einzige Änderung wäre die Stromversorgung über das Notebook und das Logging.

**Reply by OttoKarlHeinz** [reply]
[putty2.log](https://github.com/user-attachments/files/16984082/putty2.log)
Ungefähr ab der Hälfte des Mitschnittes aktualisierte die openDTU nicht mehr und ab ca. 80 Prozent fand die Aktualisierung der OpenDTU ohne Neustart wieder statt.
